### PR TITLE
Set up OAuth2

### LINKED ISSUE #55: Set up OAuth2
## Description

- Add OAuth2 features: clients, tokens
- Add basic authentication when creating OAuth2 client

## Acceptance Criteria

Able to create OAuth2 client with Basic authentication

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolve #55 

## What happened 👀

As we will use OAuth2 to authenticate the application via API, so we need to set up OAuth2 features

## Insight 📝

- Use [go-oauth2/oauth2](github.com/go-oauth2/oauth2)
- Set up OAuth server
- Create an endpoint to create a new OAuth client and protect it with Basic authentication

## Proof Of Work 📹

Postman

### CODE REVIEW

File: controllers/api/v1/oauth_client.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	oauthservice "github.com/hoangmirs/go-scraper/services/oauth"
+
+	"github.com/beego/beego/v2/core/logs"
+)
+
+type OAuthClient struct {
+	baseController
+}
+
+
```
Comment (olivierobert) [inline controllers/api/v1/oauth_client.go:14]: Extra line to remove ✂️

**Review by olivierobert — APPROVED** [review]
LGTM except the test which needs to be fixed 💪

**Review by junan — APPROVED** [review]

File: services/oauth/oauth_client.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/go-oauth2/oauth2/v4/models"
+	"github.com/google/uuid"
+)
+
+func GenerateClient() (*models.Client, error) {
+	clientID := uuid.New().String()
+	clientSecret := uuid.New().String()
+
+	client := &models.Client{
```
Comment (junan) [inline services/oauth/oauth_client.go:12]: How about to pass `UserID` also to associate with the client with the user/.

Comment (Lahphim) [reply services/oauth/oauth_client.go:12]: @junan It's a different thing 😛
This one just only create a client

**Review by Lahphim — APPROVED** [review]

**Pull request merged**